public/app/percona/dbaas/hooks/useKubernetesList.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useCancelToken } from '../../shared/components/hooks/cancelToken.hook';
+import { fetchKubernetesAction } from '../../shared/core/reducers';
+import { getKubernetes as getKubernetesSelector } from '../../shared/core/selectors';
+import {
+  CHECK_OPERATOR_UPDATE_CANCEL_TOKEN,
+  GET_KUBERNETES_CANCEL_TOKEN,
+} from '../components/Kubernetes/Kubernetes.constants';
+import { Kubernetes } from '../components/Kubernetes/Kubernetes.types';
+
+export const useKubernetesList = (): [Kubernetes[] | undefined, boolean] => {
+  const [generateToken] = useCancelToken();
+  const dispatch = useDispatch();
+  const { result: kubernetes, loading } = useSelector(getKubernetesSelector);
+
+  useEffect(() => {
+    dispatch(
+      fetchKubernetesAction({
+        kubernetes: generateToken(GET_KUBERNETES_CANCEL_TOKEN),
+        operator: generateToken(CHECK_OPERATOR_UPDATE_CANCEL_TOKEN),
+      })
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return [kubernetes, loading];
+};
+
+export const useUpdateOfKubernetesList = (): [Kubernetes[] | undefined, boolean] => {
```

Comment:
If I'm reading it right, it dispatches `fetchKubernetesAction` if it isn't fetched yet?

Reply:
I needed 2 hooks. one - useKubernetesList works usual way. when you load your component it always send fetch. The other variant - useUpdateOfKuberneteslist send dispatch once. Maybe I named it incorrect way, but the point is to send a request only if we have not previously received kubernetes values...
Individual pages - create, edit are usually opened from the interface. At this point, we have already loaded all the cluster values. In case the user wants to refresh the page and reload it, for example ctrl+shift+r - the system will pull up the data. But at the same time, it will not make an extra request if the user simply goes from one page to another

Reply:
Makes sense, thanks for the explanation.